blobfuse/blobfuse.h

```diff
@@ -37,8 +37,34 @@
 
 using namespace microsoft_azure::storage;
 
-// Internal class used for locking, see fileapis.cpp.
-class file_lock_map;
+// We use two different locking schemes to protect files / blobs against data corruption and data loss scenarios.
+// The first is an in-memory std::mutex, the second is flock (Linux).  Each file path gets its own mutex and flock lock.
+// The in-memory mutex should only be held while control is in a method that is directly communicating with Azure Storage.
+// The flock lock should be held continuously, from the time that the file is opened until the time that the file is closed.  It should also be held during blob download and upload.
+// Blob download should hold the flock lock in exclusive mode.  Read/write operations should hold it in shared mode.
+// Explanations for why we lock in various places are in-line.
+
+// This class contains mutexes that we use to lock file paths during blob upload / download / delete.
+// Each blob / file path gets its own mutex.
+// This mutex should never be held when control is not in an open(), flush(), or unlink() method.
+class file_lock_map
```

Comment:
i put several comments on this class in other CR, do you want me to dup them here as all?

Reply:
let's address this outside of this PR.